### PR TITLE
Update docs for UCC and add excitation generators

### LINKED ISSUE #307: Inconsistent `UCC` documentation
### What is the expected enhancement?

The [documentation of `UCC`](https://qiskit.org/documentation/nature/stubs/qiskit_nature.circuit.library.UCC.html) refers to the documentation of `qiskit_nature.circuit.library.ansatzes.utils.fermionic_excitation_generator` [here](https://github.com/Qiskit/qiskit-nature/blob/main/qiskit_nature/circuit/library/ansatzes/ucc.py#L76-L78).
However, the documentation of this utility method is not accessible via the public documentation.

### PR DESCRIPTION
**Pull request by woodsp-ibm**
Fixes #307

Update documentation to add excitation generator methods and fix refs etc to these.

The ansatzes.utils had sphinx docs in the init file but it was not linked into the main docs. So I added this as a link to circuit.library where initial states and ansatzes are documented (the are rolled up to there and exported from there). Removed unused sphinx docs on the anstazes and initial states init files since these are not used (all are documented in the parent).

The issue, though tagged as a documentation type, hence this documentation fix, has other comments around code improvements on validation of excitation lists from users callback etc. This is some feature enhancement that can be separated off into its own issue for future improvement if that's still wanted/needed. This PR just addresses the documentation and methods that were referenced by the docs were not part of the docs, and should have been.

### CODE REVIEW

File: qiskit_nature/circuit/library/ansatzes/ucc.py [hunk 1]
```diff
@@ -33,10 +33,11 @@
 class UCC(EvolvedOperatorAnsatz):
     r"""The Unitary Coupled-Cluster Ansatz. For more information, see [1].
 
-    This Ansatz is an :class:`~.EvolvedOperatorAnsatz` given by :math:`e^{T - T^{\dagger}}` where
+    This Ansatz is an `EvolvedOperatorAnsatz` given by :math:`e^{T - T^{\dagger}}` where
     :math:`T` is the *cluster operator*. This cluster operator generally consists of excitations
```
Comment (mrossinek) [inline qiskit_nature/circuit/library/ansatzes/ucc.py:37]:
```suggestion
    :math:`T` is the *cluster operator*. This cluster operator generally consists of excitation
```

Comment (pbark) [reply qiskit_nature/circuit/library/ansatzes/ucc.py:37]: @mrossinek This cluster operator generally consists of a sum of excitations. I do not understand your suggestion.

**Review by pbark — APPROVED** [review]

File: qiskit_nature/circuit/library/ansatzes/ucc.py (hunk 1 above)
Comment (mrossinek) [reply qiskit_nature/circuit/library/ansatzes/ucc.py:37]: It says: `excitations operators` which does not work. I suggest it to be `excitation operators`

Comment (pbark) [reply qiskit_nature/circuit/library/ansatzes/ucc.py:37]: ok now I see. thanks

**Review by pbark — APPROVED** [review]

**Review by mrossinek — APPROVED** [review]
LGTM, thanks @woodsp-ibm!

**Pull request merged**